tests/js/core/widget/core/Drawer/Drawer.js

```diff
@@ -0,0 +1,684 @@
+				assert.ok(true, "_setActive has been called");
+				assert.ok(bool !== undefined, "argument has been provided");
+			});
+
+			helpers.stub(widget, "trigger", function (eventName, details) {
+				assert.ok(true, "trigger has been called");
+				assert.ok(!!details, "details has been provided");
+			});
+
+			helpers.stub(widget, "close", function () {
+				assert.ok(true, "close has been called");
+			});
+
+			// Case open
+			widget._state = "settling";
+			widget._settlingType = "opened";
+
+			widget._onTransitionEnd();
+
+			assert.equal(widget._state, "opened", "widget state has been changed to 'opened'");
+
+			// Case close
+			widget._state = "settling";
+			widget._settlingType = "sliding";
+			widget._drawerOverlay = document.createElement("div");
+
+			widget._onTransitionEnd();
+
+			assert.equal(widget._state, "closed", "widget state has been changed to 'closed'");
+			assert.equal(widget._drawerOverlay.style.visibility, "hidden", "widget overlay hidden");
+		});
+
+
+		test("Test of open drawer", 21, function (assert) {
+			var widget = new Drawer(),
+				element = document.getElementById("drawer-1");
+
+			widget._state = "not opened";
+			widget._settlingType = "not defined";
+			widget.element = element;
+			widget._ui = {
+				drawerOverlay: document.createElement("div")
+			};
+
+			// case left
+			widget.options = {
+				position: "left"
+			};
+			helpers.stub(widget, "_translate", function (x, y, duration) {
+				assert.ok(true, "_translate has been called");
+				assert.equal(x, 0, "x has been provided as 0");
+				assert.equal(y, 0, "y has been provided as 0");
+				assert.equal(duration, 1000, "duration has been provided as 1000");
+			});
+			widget.open(1000);
+
+			// common for all cases
+			assert.equal(widget._state, "settling", "widget state has been changed to 'settling'");
+			assert.equal(widget._settlingType, "opened", "widget _settlingType has been changed to 'opened'");
+			assert.equal(widget._ui.drawerOverlay.style.visibility, "visible", "widget overaly visible");
+			assert.ok(!element.classList.contains("ui-drawer-close"), "widget doesn't have close class");
+			assert.ok(element.classList.contains("ui-drawer-open"), "widget has open class");
+
+			// case up
+			widget._state = "not opened";
+			widget.options = {
+				position: "up"
+			};
+			helpers.stub(widget, "_translate", function (x, y, duration) {
+				assert.ok(true, "_translate has been called");
+				assert.equal(x, 0, "x has been provided as 0");
+				assert.equal(y, 0, "y has been provided as 0");
+				assert.equal(duration, 500, "duration has been provided as 500");
+			});
+			widget.open(500);
+
+			// case right
+			widget._state = "not opened";
+			widget.options = {
+				position: "right",
+				width: window.innerWidth
+			};
+			helpers.stub(widget, "_translate", function (x, y, duration) {
+				assert.ok(true, "_translate has been called");
+				assert.equal(x, 0, "x has been provided as window.innerWidth");
```

Comment:
Is the '0' correct value for checking window.innerWidth?

Reply:
I'll change this.

Reply:
Done.